granulate_utils/linux/cgroups/base_cgroup.py

```diff
@@ -5,17 +5,61 @@
 
 import os
 from pathlib import Path
-from typing import List, Tuple
+from typing import List, Mapping, Optional, Tuple
 
 from psutil import NoSuchProcess
 
-CGROUPFS = Path("/sys/fs/cgroup")
-SUBSYSTEMS = {"memory", "cpu"}
-
+from granulate_utils.exceptions import AlreadyInCgroup
+from granulate_utils.linux.mountinfo import iter_mountinfo
+from granulate_utils.linux.ns import resolve_host_root_links
 
-class AlreadyInCgroup(Exception):
-    def __init__(self, subsystem: str, cgroup: str) -> None:
-        super().__init__(f"{subsystem!r} subsytem is already in a predefined cgroup: {cgroup!r}")
+# TODO: This is not always correct. For example, on Amazon Linux 1 this should be /cgroup.
+CGROUPFS = Path("/sys/fs/cgroup")
```

Comment:
Let's take the effort here now to find it dynamically?

Reply:
done